planner/core/rule_join_reorder.go

```diff
@@ -181,12 +177,30 @@ func (s *joinReOrderSolver) optimizeRecursive(ctx sessionctx.Context, p LogicalP
 		joinGroupNum := len(curJoinGroup)
 		useGreedy := joinGroupNum > ctx.GetSessionVars().TiDBOptJoinReorderThreshold || !isSupportDP
 
-		if hintInfo != nil && hintInfo.leadingJoinOrder != nil {
+		leadingHintNum := len(hintInfo)
+		var leadingHintInfo *tableHintInfo
+		if leadingHintNum > 0 {
+			hasDiffLeadingHint := false
+			leadingHintInfo = hintInfo[0]
+			// One join group has one leading hint at most. Check whether there are different join order hints.
+			for i := 1; i < leadingHintNum; i++ {
+				if hintInfo[i] != hintInfo[i-1] {
+					hasDiffLeadingHint = true
```

Comment:
Why check it here again, isn't it duplicated with `logical_plan_builder.go:L3633`?

Reply:
`logical_plan_builder.go:L3633` check the leading hints in the same query block.

For here, some leading hints in different query block can appear in the same join group. Such as, `select /*+ leading(t3) */ * from (select /*+ leading(t1) */ t2.b from t1 join t2 on t1.a=t2.a) t4 join t3 on t4.b=t3.b`